# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Lowering.cpp
# Commit: fdbb1c215d78
# Full Hash: fdbb1c215d783a85c27d71ed8bb832690962c948
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2016-03-28 20:04:22
# Regressor Bug: 1165052
# File Overlap Count: 1
# Description:
#   Bug 1165052 - Part 4: Inline IsConstructor intrinsic. r=jandem
# ==============================================================================

diff -r 5e02cbf51e45 -r fdbb1c215d78 js/src/jit/Lowering.cpp
--- a/js/src/jit/Lowering.cpp	Wed Dec 23 12:46:35 2015 +0900
+++ b/js/src/jit/Lowering.cpp	Mon Nov 16 09:01:37 2015 +0900
@@ -3965,6 +3965,14 @@
     define(new(alloc()) LIsCallable(useRegister(ins->object())), ins);
 }
 
+void
+LIRGenerator::visitIsConstructor(MIsConstructor* ins)
+{
+    MOZ_ASSERT(ins->object()->type() == MIRType_Object);
+    MOZ_ASSERT(ins->type() == MIRType_Boolean);
+    define(new(alloc()) LIsConstructor(useRegister(ins->object())), ins);
+}
+
 static bool
 CanEmitIsObjectAtUses(MInstruction* ins)
 {